roqoqo-qryd/tests/integration/api_backend.rs

```
@@ -12,260 +12,435 @@
-        assert!(!bits.is_empty());
-        for line in bits["ro"].iter() {
-            println!("{:?}", line);
-        }
+    let server = MockServer::start();
+    let qryd_job_status_in_progress = QRydJobStatus {
+        status: "in progress".to_string(),
+        msg: "the job is still in progress".to_string(),
+    };
+    let qryd_job_status_completed = QRydJobStatus {
+        status: "completed".to_string(),
+        msg: "the job has been completed".to_string(),
+    };
+    let result_counts = ResultCounts {
+        counts: HashMap::from([("0x1".to_string(), 100), ("0x4".to_string(), 20)]),
+    };
+    let qryd_job_result_completed = QRydJobResult {
+        data: result_counts,
+        time_taken: 0.23,
+        noise: "noise".to_string(),
+        method: "method".to_string(),
+        device: "QrydEmuSquareDevice".to_string(),
+        num_qubits: 4,
+        num_clbits: 4,
+        fusion_max_qubits: 4,
+        fusion_avg_qubits: 4.0,
+        fusion_generated_gates: 100,
+        executed_single_qubit_gates: 50,
+        executed_two_qubit_gates: 50,
+    };
+    let mock_post = server.mock(|when, then| {
+        when.method("POST");
+        then.status(201).header(
+            "Location",
+            format!("http://127.0.0.1:{}/DummyLocation", server.port()),
+        );
+    });
+    let mut mock_status0 = server.mock(|when, then| {
+        when.method("GET").path("/DummyLocation/status");
+        then.status(200).json_body_obj(&qryd_job_status_in_progress);
+    });
+    let mock_result = server.mock(|when, then| {
+        when.method("GET").path("/DummyLocation/result");
+        then.status(200).json_body_obj(&qryd_job_result_completed);
+    });
+
+    let number_qubits = 6;
+    let device = QrydEmuSquareDevice::new(Some(2), Some(0.23));
+    let qryd_device: QRydAPIDevice = QRydAPIDevice::from(&device);
+    let api_backend_new =
+        APIBackend::new(qryd_device, None, None, Some(server.port().to_string())).unwrap();
```

The same as in the python interface tests, we always want to use the non-mocked backend when the access token is present and fall back to the mocked one when it is not.